pkg/controller/garbagecollector/patch.go

```diff
@@ -32,16 +32,20 @@ import (
 // getMetadata tries getting object metadata from local cache, and sends GET request to apiserver when
 // local cache is not available or not latest.
 func (gc *GarbageCollector) getMetadata(apiVersion, kind, namespace, name string) (metav1.Object, error) {
-	apiResource, _, err := gc.apiResource(apiVersion, kind)
+	apiResource, namespaced, err := gc.apiResource(apiVersion, kind)
 	if err != nil {
 		return nil, err
 	}
+	namespace = resourceDefaultNamespace(namespaced, namespace)
```

Comment:
also check this here:

```go
	if namespaced && len(namespace) == 0 {
		// the type is namespaced, but we have no namespace coordinate.
		// the only way this can happen is if a cluster-scoped object referenced this type as an owner.
		return nil, namespacedOwnerOfClusterScopedObjectErr
	}
```